Tidy up roles pages (#5657)
EnterpriseDB/docs: product_docs/docs/pgd/5/security/pgd-predefined-roles.mdx

---
title: PGD predefined roles
description: Describes predefined roles in PGD
---

PGD predefined roles are created when the BDR extension is installed. After BDR
extension is dropped from a database, the roles continue to exist. You need to
drop them manually if dropping is required. 

### bdr_superuser
 

> This is a role for an admin user that can manage anything PGD related. It allows you to separate management of the database and table access. Using it allows you to have a user that can manage the PGD without giving them pg_superuser privileges.

Reviewer: ```suggestion
This is a role for an admin user that can manage anything PGD related. It allows you to separate management of the database and table access. Using it allows you to have a user that can manage the PGD cluster without giving them PostgreSQL superuser privileges.
```

Reviewer: There's no `pg_superuser` role.

#### Privileges

-   ALL PRIVILEGES ON ALL TABLES IN SCHEMA BDR
-   ALL PRIVILEGES ON ALL ROUTINES IN SCHEMA BDR

### bdr_read_all_stats

> This role provides read access to most if the tables, views and functions that may be needed by users or applications which wish to observe the statistics and state of the PGD cluster. 

Reviewer: ```suggestion
This role provides read access to most of the tables, views and functions that may be needed by users or applications which wish to observe the statistics and state of the PGD cluster. 
```

#### Privileges

> SELECT privilege on:

Reviewer: ```suggestion
`SELECT` privilege on:
```

### bdr_monitor

This role provides read access to any tables, views and functions that may be needed by users or applications which monitor the PGD cluster. It includes all the privileges of the [`bdr_read_all_stats`](#bdr_read_all_stats) role.

#### Privileges

All privileges from [`bdr_read_all_stats`](#bdr_read_all_stats) plus the following additional privileges:

> SELECT privilege on:

Reviewer: ```suggestion
`SELECT` privilege on:
```

-    `bdr.monitor_version_details_on_all_nodes`
-    `bdr.node_group_member_info`

### bdr_application

This role is designed for applications which require access to PGD features, objects and functions such as sequences, CRDT datatypes, CAMO status functions or trigger management functions.

#### Privileges

> EXECUTE privilege on:

Reviewer: ```suggestion
`EXECUTE` privilege on:
```
